pkg/datasource/accessors.go

```diff
@@ -445,6 +472,126 @@ func (a *fieldAccessor) Float64(data Data) (float64, error) {
 	return math.Float64frombits(i), nil
 }
 
+// Array functions
+// to be discussed: these methods use a slow copying method to return the arrays
+// It can also be done using for the unsafe package, like:
+// return unsafe.Slice((*uint64)(unsafe.Pointer(&val[0])), len(val)/8), nil
+// I _think_ it's okay, but if there are any reasons against it, please let me know.
```

Comment:
`copy()` would only work for the uint8 variant as all other array types are not compatible. As those array functions also respect the `byteOrder`, the unsafe version could only be used if `byteOrder == NativeEndian` (which is probably almost always the case).

I did not profile yet, but a cast (which the unsafe version basically is) would be much faster than copying memory for sure.

Reply:
Btw, those helper functions are provided to complete the API. There's always the manual way as an alternative if you require speed - which is exactly what I've done in the otel operator.

Reply:
> but without casting it to anything other than byte

Not sure I can follow you there. If the helper is Uint16Array(), it should return `[]uint16`, right?

> Calculating the length with types > sizeof() > 1 is error prone.

Wdym?

The biggest reason not to do it with unsafe is that `data` and its fields must not be accessed after returning from the subscription callback. Returning an array that is directly based on the underlying memory (instead of the safe primitives from the other helpers) feels more prone to be misused.

Reply:
> Wdym?

I think I misread the comment and more. I thought `unsafe.Slice((*uint64)(unsafe.Pointer(&val[0])), len(val)/8)` was meant to be a general way to copy any kind of array.
After rereading:
1. unsafe.Slice doesn't copy anything
2. This was meant for the uint64 array only

If we cannot guarantee the safety, either through safe primitives or custom types, we should copy them.